crates/uv/src/commands/project/run.rs

```diff
@@ -996,9 +997,27 @@ pub(crate) async fn run(
     // signal handlers after the command completes.
     let _handler = tokio::spawn(async { while tokio::signal::ctrl_c().await.is_ok() {} });
 
-    let status = handle.wait().await.context("Child process disappeared")?;
+    #[cfg(unix)]
+    {
+        use tokio::select;
+        use tokio::signal::unix::{signal, SignalKind};
+        let mut term_signal = signal(SignalKind::terminate())?;
+        loop {
+            select! {
+                _ = handle.wait() => {
+                    break
+                },
+
+                // `SIGTERM`
+                _ = term_signal.recv() => {
+                    let _ = terminate_process(&mut handle);
+                }
+            };
+        }
+    }
 
     // Exit based on the result of the command
+    let status = handle.wait().await?;
```

Comment:
Is it "fine" / necessary for us to be calling `.wait()` twice?

Reply:
Not a rust expert would appreciate your guidance here, from what I see inside of `wait()` it's not expensive to call it twice and it makes the logic in the select! block easier to read. If I do something like let status = select! {...} each brand needs to return the same type, but `terminate_process` doesn't actually return anything, so I'd need to fake a return which also seems unnecessary. wdyt?

Reply:
It looks like it _is_ guaranteed to return the same result both times.

Reply:
Nice, agreed this looks better 👍